packages/babel-parser/src/parser/statement.js

```diff
@@ -2005,13 +2005,29 @@ export default class StatementParser extends ExpressionParser {
     // import '...'
     node.specifiers = [];
     if (!this.match(tt.string)) {
+      // check if we have a default import like
+      // import React from "react";
       const hasDefault = this.maybeParseDefaultImportSpecifier(node);
+      /* we are checking if we do not have a default import, then it is obvious that we need named imports
+       * import { get } from "axios";
+       * but if we do have a default import
+       * we need to check if we have a comma after that and
+       * that is where this || this.eat comes into play
+       */
       const parseNext = !hasDefault || this.eat(tt.comma);
+      // if we do have to parse the next set of specifiers, we first check for start imports
+      // import React, * from "react";
       const hasStar = parseNext && this.maybeParseStarImportSpecifier(node);
+      // now we check that if we need to parse the next imports
+      // but only if he/she is not importing * which means all
       if (parseNext && !hasStar) this.parseNamedImportSpecifiers(node);
       this.expectContextual("from");
     }
     node.source = this.parseImportSource();
+    const attributes = this.maybeParseModuleAttributes();
+    if (Array.isArray(attributes)) {
+      node.attributes = attributes;
+    }
```

Comment:
As a micro-optimization (I know that this logic was suggested by me :stuck_out_tongue:) we could do
```js
if (attributes) node.attributes = attributes;
```
which looks more like "if attributes are present, then set them" and avoids needing to know that it is an array when skimming throughout this function.

Reply:
Shouldn't the module attribute be a map instead of an array? I would prefer rejecting duplicated attributes to avoid confusion.

Reply:
> Shouldn't the module attribute be a map instead of an array?

Since the syntax isn't `{ foo: bar }`, I'm not sure that I'd expect a map.

Reply:
The key of a module attribute must be it's own AST node, which can have comments and location info.
We are also using an array for imported bindings (i.e. `{ x as y, x as w }`).

Reply:
Ok, that's fine to me too